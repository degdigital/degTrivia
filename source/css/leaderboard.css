@import "variables.css";
@import "basics/00-reset.css";
@import "basics/headings.css";
@import "basics/table.css";
@import "basics/text.css";
@import "basics/typography.css";
@import "components/pregame/next-game";
@import "components/tv-screen/tv-screen";
@import "templates/layout.css";
@import "templates/columns.css";