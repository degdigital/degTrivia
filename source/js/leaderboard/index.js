// Firebase
import firebase from '@firebase/app';
import firebaseConfig from '../config/firebaseConfig.js';

// Services
import dbService from '../services/dbService.js';
import playerService from '../services/playerService.js';

// Screens
import login from '../admin/screens/login.js';
import screenRotator from './components/screenRotator.js';

const leaderboardTV = function(el) {

	let db;
	let loginInst;

	function setUpListeners() {
		playerService.getAuth().onAuthStateChanged(onAuthStateChange);
	}

	async function onAuthStateChange(user) {
		if (user) {
			const isAdmin = await db.ref(`admins/${user.uid}`).once('value').then(snapshot => snapshot.exists() && snapshot.val() === true);
			if (isAdmin === true) {
				screenRotator(el);
			} else {
				loginInst.renderForm();
			}
		} else {
			loginInst.renderForm();
		}
	}

	function init() {
		firebase.initializeApp(firebaseConfig);
		dbService.init();
		playerService.init();

		db = dbService.getDb();
		loginInst = login(el);
		
		setUpListeners();
	}

	init();
};

const rootEl = document.querySelector('.app');
leaderboardTV(rootEl);
