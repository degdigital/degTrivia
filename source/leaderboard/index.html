<!DOCTYPE html>
<html lang="en-us">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <title>DEG Trivia Leaderboard</title>
        <link rel="stylesheet" type="text/css" href="/css/leaderboard.css">
    </head>
    <body>
        <div id="app"></div>

        <!-- for modern browsers -->
        <script type="module" src="/js/leaderboard-bundle.js"></script>

        <!-- for older browsers -->
        <script nomodule src='/js/leaderboard-bundle-nomodules.js'></script>        
    </body>
</html>
