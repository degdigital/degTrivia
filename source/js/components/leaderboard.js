import {replaceContent} from '../utils/domUtils.js';
import dbService from '../services/dbService.js';

function loadLeaderboardData() {
    return dbService.getLeaderboardData();
}

function renderTableBody(leaderData) {
    return leaderData.map((leader, index) => (
        `<tr class="table__row table__data-row">
            <td class="table__data-cell">${index + 1}</td>
            <td class="table__data-cell">${leader.name}</td>
            <td class="table__data-cell">${leader.score}</td>
        </tr>`
    )).join('');
}

function renderLeaderboard(data) {
    let returnMarkup;
    if (data) {
        returnMarkup = `
            <table class="table">
                <caption class="table__caption">Leaders</caption>
                <colgroup>
                    <col class="table__col table__col--small"></col>
                    <col class="table__col"></col>
                    <col class="table__col"></col>
                </colgroup>
                <tbody class="table__table-body">${renderTableBody(data)}</tbody>
            </table>
        `;
    } else {
        returnMarkup = '<div>No leaderboard data to display.</div>'
    }
   
    return returnMarkup;
}

async function render(containerElement) {
    const leaderboardData = await loadLeaderboardData();
    replaceContent(containerElement, renderLeaderboard(leaderboardData.game));
}

export default function() {
    return {
        renderToElement: render
    }
}