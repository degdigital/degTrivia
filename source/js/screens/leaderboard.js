import {replaceContent} from '../utils/domUtils.js';

const leaderboard = function({element}) {

	function render() {
		replaceContent(element, `
			Leaderboard
		`);
	}

	return {
		render
	};

};

export default leaderboard;