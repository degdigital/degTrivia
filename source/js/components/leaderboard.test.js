import leaderboard from './leaderboard.js';

jest.mock('../services/dbService');

let element;

describe('leaderboard', () => {

	beforeEach(() => {
		element = document.createElement('div');
		document.body.appendChild(element);
	})

    it('should render', async () => {
		await leaderboard().renderToElement(element);
		expect(element).toMatchSnapshot();
    });
});