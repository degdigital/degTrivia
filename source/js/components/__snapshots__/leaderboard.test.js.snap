// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`leaderboard should render 1`] = `
<div>
  
            
  <table
    class="table"
  >
    
                
    <caption
      class="table__caption"
    >
      Leaders
    </caption>
    
                
    <colgroup>
      
                    
      <col
        class="table__col table__col--small"
      />
      
                    
      <col
        class="table__col"
      />
      
                    
      <col
        class="table__col"
      />
      
                
    </colgroup>
    
                
    <tbody
      class="table__table-body"
    >
      <tr
        class="table__row table__data-row"
      >
        
            
        <td
          class="table__data-cell"
        >
          1
        </td>
        
            
        <td
          class="table__data-cell"
        >
          Anna
        </td>
        
            
        <td
          class="table__data-cell"
        >
          1500000
        </td>
        
        
      </tr>
      <tr
        class="table__row table__data-row"
      >
        
            
        <td
          class="table__data-cell"
        >
          2
        </td>
        
            
        <td
          class="table__data-cell"
        >
          Aaron
        </td>
        
            
        <td
          class="table__data-cell"
        >
          140
        </td>
        
        
      </tr>
      <tr
        class="table__row table__data-row"
      >
        
            
        <td
          class="table__data-cell"
        >
          3
        </td>
        
            
        <td
          class="table__data-cell"
        >
          Ryan
        </td>
        
            
        <td
          class="table__data-cell"
        >
          42
        </td>
        
        
      </tr>
    </tbody>
    
            
  </table>
  
        
</div>
`;
